src/test/java/org/broadinstitute/hellbender/engine/dataflow/DataflowTestData.java

```diff
@@ -0,0 +1,215 @@
+                KV.of(reads.get(1), variants.get(0)),
+                KV.of(reads.get(1), variants.get(1)),
+                KV.of(reads.get(2), variants.get(2)),
+                KV.of(reads.get(3), variants.get(3)),    // The read and variant span two variant shards, that's
+                KV.of(reads.get(3), variants.get(3))     // why there are two of them (2,3).
+        );
+
+        Iterable<Variant> variant10 = Lists.newArrayList(kvReadVariant.get(1).getValue(), kvReadVariant.get(0).getValue());
+        Iterable<Variant> variant01 = Lists.newArrayList(kvReadVariant.get(0).getValue(), kvReadVariant.get(1).getValue());
+        Iterable<Variant> variant2 = Lists.newArrayList(kvReadVariant.get(2).getValue());
+        Iterable<Variant> variant3 = Lists.newArrayList(kvReadVariant.get(3).getValue());
+
+        kvReadiVariant = Arrays.asList(
+                KV.of(kvReadVariant.get(0).getKey(), variant10),
+                KV.of(kvReadVariant.get(2).getKey(), variant2),
+                KV.of(kvReadVariant.get(3).getKey(), variant3)
+        );
+
+        // We also need alternative ordering of the variants because of the fragility of DaflowAssert's containInAnyOrder.
+        kvReadiVariant2 = Arrays.asList(
+                KV.of(kvReadVariant.get(0).getKey(), variant01),
+                KV.of(kvReadVariant.get(2).getKey(), variant2),
+                KV.of(kvReadVariant.get(3).getKey(), variant3)
+        );
+
+        kvReadContextData = Arrays.asList(
+                KV.of(kvReadsRefBases.get(0).getKey(), new ReadContextData(kvReadsRefBases.get(0).getValue(), Lists.newArrayList())),
+                KV.of(kvReadsRefBases.get(1).getKey(), new ReadContextData(kvReadsRefBases.get(1).getValue(), kvReadiVariant.get(0).getValue())),
+                KV.of(kvReadsRefBases.get(2).getKey(), new ReadContextData(kvReadsRefBases.get(2).getValue(), kvReadiVariant.get(1).getValue())),
+                KV.of(kvReadsRefBases.get(3).getKey(), new ReadContextData(kvReadsRefBases.get(3).getValue(), kvReadiVariant.get(2).getValue()))
+        );
+    }
+
+    public GATKRead makeRead(KV<Integer, Integer> startLength, int i, Class<?> clazz) {
+        return makeRead(startLength.getKey(), startLength.getValue(), i, clazz);
+    }
+
+    public GATKRead makeRead(int start, int length, int i, Class<?> clazz) {
+        if (clazz == Read.class) {
+            return ArtificialReadUtils.createGoogleBackedReadWithUUID(new UUID(0, i), Integer.toString(i), start, length);
+        } else if (clazz == SAMRecord.class) {
+            return ArtificialReadUtils.createSamBackedReadWithUUID(new UUID(0, i), Integer.toString(i), start, length);
+        } else {
+            throw new GATKException("invalid GATKRead type");
+        }
+    }
+
+    private SimpleInterval makeInterval(KV<Integer, Integer> startLength) {
+        return new SimpleInterval("1", startLength.getKey(), startLength.getKey() + startLength.getValue() - 1);
```

Comment:
It would be good to have something on another chromosome to show that it doesn't accidentally aggregate across chromosomes.